### PR TITLE
🧩 ERC-7412 Support

### PR DESCRIPTION
**Pull request by JaredBorders**
Support calls to `ERC-7412.fulfillOracleQuery` function from the SMv3 Engine

## Description
* Implement `EIP7412.sol`
* Add `ERC2771Context` to `Engine.sol`
* Replace `msg.sender` with `ERC2771Context._msgSender()`
* Add `_trustedForwarder` parameter to the `Engine.constructor`
* Test `EIP7412.sol`
* Deploy SMv3 to Base Goerli
* Update gas-snapshot
* Update lcov
* Update `.env-example` (PRIVATE_KEY must now start with `0x` else parsing will fail within deploy script)

## Related issue(s)
N/A

## Motivation and Context
These changes are required to interact with Synthetix Perps v3 after it adopted ERC-7412

### CODE REVIEW

**Review by JChiaramonte7 — APPROVED** [review]
LGTM, seems pretty straight forward.

My only question: is the intent by not restricting function calls to the trusted forwarder, to enable our own internal multicall to be used in the future?

File: script/utils/parameters/OptimismParameters.sol [hunk 1]
```diff
@@ -11,4 +11,7 @@ contract OptimismParameters {
         0xb2F30A7C980f052f02563fb518dcc39e6bf38175;
 
     address public constant PYTH = 0xff1a0f4744e8582DF1aE09D5611b887B6a12925C;
+
+    /// @custom:todo Optimism doesn't have a trusted forwarder yet
+    address public constant TRUSTED_FORWARDER = address(0);
```
Comment (tommyrharper) [inline script/utils/parameters/OptimismParameters.sol:16]: **Question:** Are we not going to use this contract yet then? We will just wait until the trusted forwarder is deployed on OP before we deploy our new SMv3?

File: test/EIP7412.t.sol [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        (bool success,) = address(eip7412).call{value: 5 wei}(
+            abi.encodeWithSelector(
+                EIP7412.fulfillOracleQuery.selector,
+                mock_eip7412_implementer,
+                signedOffchainData
+            )
+        );
+
+        assertTrue(success);
+        assertLt(address(this).balance, preBalance);
```
Comment (tommyrharper) [inline test/EIP7412.t.sol:29]: **Question:** Why is `address(this).balance == preBalance`? How come it is not `5 wei` less after calling `fulfillOracleQuery` on the `eip7412` contract? As far as I can tell, the `eip7412` will simply accept the `5 wei` and do nothing with it.

**Review by tommyrharper — APPROVED** [review]
LGTM 👍

File: script/utils/parameters/OptimismParameters.sol (hunk 1 above)
Comment (JaredBorders) [reply script/utils/parameters/OptimismParameters.sol:16]: We will wait until the trusted forwarder is on OP before deploying

**Pull request merged**